src/api/define.js

```diff
@@ -1,48 +1,49 @@
-import { name as $name } from '../util/symbols';
 import Component from './component';
 import uniqueId from '../util/unique-id';
 import root from 'window-or-global';
 
 export default function (...args) {
-  const { customElements } = root;
+  const { customElements HTMLElement } = root;
   let [ name, Ctor ] = args;
 
   if (!customElements) {
     throw new Error('Skate requires native custom element support or a polyfill.');
   }
 
-  // Support passing an anonymous definition.
-  if (args.length === 1) {
-    // We are checking string for now, but once we remove the ability to pass
-    // an object literal, we can change this to check "function" and invert the
-    // blocks of logic.
-    if (typeof name === 'string') {
-      throw new Error('When passing only one argument to define(), it must be a custom element constructor.');
-    } else {
-      Ctor = name;
-      name = uniqueId();
-    }
+  // DEPRECATED remove when removing the "name" argument.
+  if (DEBUG && args.legnth === 2) {
+    console.warn('The "name" argument to define() is deprecated. Please define a `static is` property on the constructor instead.');
   }
 
-  // Ensure there's no conflicts.
-  if (customElements.get(name)) {
-    name = uniqueId(name);
+  // DEPRECATED remove when removing the "name" argument.
+  if (args.length === 1) {
```

Comment:
I tried doing this in the last updates, but I found that it was less readable, and I still needed to cater to a `Ctor` that may be an object literal. I found that it's much more clear what's deprecated and what's going on if things were in separate blocks. At the very least, once we remove the deprecations, we can remove the `let` blocks and things should be come just as simple, it'll just take until the next major release.